parser/dirs.go

```diff
@@ -88,6 +89,17 @@ func parseDir(fset *token.FileSet, dir, relPath string, filter func(os.FileInfo)
 				return nil, nil, err
 			}
 
+			// Check if this file should be part of the build
+			matched, err := context.MatchFile(dir, d.Name())
+			if err != nil {
+				errors.Add(token.Position{Filename: filename}, err.Error())
+				continue
+			}
+
+			if !matched {
+				continue
+			}
+
```

Comment:
This allows my test app to have directives such as

```go
//go:build go1.18
```

or

```go
//go:build !encore
```

Both of which I needed to get the parser working in the test scenarios I was using (for instance I have a `main.go` file I use for testing the code actually works under go1.18 but isn't part of the encore app itself)

Reply:
Ah, nice :)